pkg/detectors/twitch/twitch.go

```diff
@@ -58,37 +62,59 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 			}
 
 			if verify {
-				data := url.Values{}
-				data.Set("client_id", resIdMatch)
-				data.Set("client_secret", resMatch)
-				data.Set("grant_type", "client_credentials")
-				encodedData := data.Encode()
-				req, err := http.NewRequestWithContext(ctx, "POST", "https://id.twitch.tv/oauth2/token", strings.NewReader(encodedData))
-				if err != nil {
-					continue
-				}
-				req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
-				req.Header.Add("Content-Length", strconv.Itoa(len(data.Encode())))
-				res, err := client.Do(req)
-				if err == nil {
-					defer res.Body.Close()
-					if res.StatusCode >= 200 && res.StatusCode < 300 {
-						s1.Verified = true
-					} else {
-						// This function will check false positives for common test words, but also it will make sure the key appears 'random' enough to be a real key
-						if detectors.IsKnownFalsePositive(resMatch, detectors.DefaultFalsePositives, true) {
-							continue
-						}
-					}
-				}
+				client := s.getClient()
+
+				isVerified, verificationErr := verifyTwitch(ctx, client, resMatch, resIdMatch)
+				s1.Verified = isVerified
+				s1.VerificationError = verificationErr
 			}
 
+			// This function will check false positives for common test words, but also it will make sure the key appears 'random' enough to be a real key
+			if !s1.Verified && detectors.IsKnownFalsePositive(resMatch, detectors.DefaultFalsePositives, true) {
+				continue
+			}
 			results = append(results, s1)
 		}
 	}
 	return results, nil
 }
 
+func (s Scanner) getClient() *http.Client {
+	if s.client != nil {
+		return s.client
+	}
+	return defaultClient
+}
+
+func verifyTwitch(ctx context.Context, client *http.Client, resMatch string, resIdMatch string) (bool, error) {
+	data := url.Values{}
+	data.Set("client_id", resIdMatch)
+	data.Set("client_secret", resMatch)
+	data.Set("grant_type", "client_credentials")
+	encodedData := data.Encode()
+
+	req, err := http.NewRequestWithContext(ctx, "POST", verifyURL, strings.NewReader(encodedData))
+	if err != nil {
+		return false, err
+	}
+
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	req.Header.Add("Content-Length", strconv.Itoa(len(data.Encode())))
+	res, err := client.Do(req)
+	if err != nil {
+		return false, err
+	}
+
+	defer res.Body.Close()
+	if res.StatusCode == http.StatusOK {
```

Comment:
nit (optional): You could switchify this and the default case handles the final return.
```
switch res.StatusCode {
case http.StatusOK:
	return true, nil
case http.StatusBadRequest, http.StatusForbidden:
	return false, nil
default:
	return false, fmt.Errorf("unexpected http response status %d", res.StatusCode)
```